src/internal.c

```diff
@@ -11492,6 +11493,10 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                             ret = ECC_KEY_SIZE_E;
                             WOLFSSL_MSG("Peer ECC key is too small");
                         }
+
+                        /* populate curve oid */
+                        if (ssl->options.side == WOLFSSL_CLIENT_END)
```

Comment:
TLSX_ValidateSupportedCurves, called in VerifyServerSuite, choses the ECDH curve to use based on extension data.

Reply:
In the test case with static ECDH the `ssl->ecdhCurveOID` was not getting set via `TLSX_ValidateSupportedCurves`. I will test again and see if I can figure out a better solution.

Reply:
Should static ECDH then be added to ValidateSupportedCurves so all the logic is in one place?

Reply:
I did add a check to only set if it isn't already populated. I dug into this again and only the server side uses `VerifyServerSuite` and `TLSX_ValidateSupportedCurves`. For the use case the actual ECC curve isn't sent until the certificate because it's a static DH cipher suite. This location is the only place the curve OID data is passed. There is another fix above in `EccMakeKey` which resolves the curve based on peer key, but OID data isn't available. Technically this fix isn't required for function, but it does properly set the `ssl->ecdhCurveOID` data used by `wolfSSL_get_curve_name`.